proj-xcode/Classes/keychain/PA2Keychain.m

```diff
@@ -291,12 +290,7 @@ - (PA2KeychainStoreItemResult) implAddValue:(NSData*)data forKey:(NSString*)key
 	[query setValue:@YES	forKey:(__bridge id)kSecUseNoAuthenticationUI];
 	
 	// If the system version is iOS 9.0+, use Touch ID if requested (kSecAccessControlTouchIDAny), or use kNilOptions
-	SecAccessControlCreateFlags flags = kNilOptions;
-	if ([PA2Keychain canUseTouchId]) {
-		if (useTouchId) {
-			flags = kSecAccessControlTouchIDAny;
-		}
-	}
+	SecAccessControlCreateFlags flags = useTouchId ? kSecAccessControlTouchIDAny : kNilOptions;
```

Comment:
We removed call to `canUseTouchId` here - is this correct? This was there to prevent using the flag on devices without fingerprint scanner. The `useTouchId` flag represents programmers desire to use Touch ID, `canUseTouchId` checks if this is possible...

Reply:
That's correct, because it's already evaluated several lines above.

Reply:
It's funny, because you would like to save several object allocations (as comment at the beginning of the method is saying), but in fact, that redundant call did allocate several unnecessary objects... :)

Reply:
OK - Sorry, I see that now, after I opened the whole file. The old impl was some leftover check...